win32ss/gdi/eng/gradient.c

```diff
@@ -401,8 +401,8 @@ IntEngGradientFillTriangle(
               ENDLINE(v1, v2, 1);
 
               GOLINE(v2, v3, 2);
-              DOLINE(v2, v3, 2);
               FILLLINE(0, 2);
+              DOLINE(v2, v3, 2);
```

Comment:
I don't know what this change does, but should this be changed at line 450 too?
And why don't  lines 400 and 445 need it too?

Reply:
I have reworked this section of the code and there is not a one-to-one relationship between these macros.
The ```DOLINE``` provides indexing functions and the ```FILLLINE``` does the line drawing. Thanks,.